workspaces/acs/plugins/acs/src/common/QueryACSData.ts

```diff
@@ -25,11 +25,11 @@ export const QueryACSData = (deploymentName: string) => {
   const [result, setResult] = useState([]);
   const [loaded, setLoaded] = useState<boolean>(false);
   const [error, setError] = useState<boolean>(false);
+  const [backendUrl, setBackendUrl] = useState('');
 
   // Retrieve proxy url from api
   const discoveryApi = useApi(discoveryApiRef);
-  let backendUrl = '';
-  discoveryApi.getBaseUrl('proxy').then(value => (backendUrl = value));
+  discoveryApi.getBaseUrl('proxy').then(value => setBackendUrl(value));
```

Comment:
I believe we will call `setBackendUrl` on every render if we do it here. Since the useEffect checks for the string and it doesn't change, it won't trigger the `getACSData` function call. But I believe it still updates the state on every render.